File: sdks/csharp/PgChangeFeed.Client/PgChangeFeed.Client.Tests/Http/TestClientFactory.cs
using System.Net.Http;
using PgChangeFeed.Client.Http;

namespace PgChangeFeed.Client.Tests.Http;

/// <summary>
/// Builds a <see cref="PgChangeFeedHttpClient"/> wired to a
/// <see cref="FakeHttpMessageHandler"/> — every test in this project stays
/// network-free (no real server, no real socket).
/// </summary>
internal static class TestClientFactory
{
    public static readonly Uri ServerAddress = new("http://example.invalid:8080");

    public static (PgChangeFeedHttpClient Client, FakeHttpMessageHandler Handler) Create(
        Func<HttpRequestMessage, HttpResponseMessage> responder, string apiToken = "test-token")
    {
        var handler = new FakeHttpMessageHandler(responder);
        var httpClient = new HttpClient(handler);
        var options = new PgChangeFeedClientOptions(ServerAddress, apiToken);
        return (new PgChangeFeedHttpClient(httpClient, options), handler);
    }
}

File: sdks/kotlin/pgchangefeed-kotlin/src/test/kotlin/io/github/pt9912/pgchangefeed/http/TestClientFactory.kt
package io.github.pt9912.pgchangefeed.http

import io.github.pt9912.pgchangefeed.PgChangeFeedClientOptions
import java.net.URI

/**
 * Builds a [PgChangeFeedHttpClient] wired to a [FakeHttpTransport] via the
 * `internal` transport constructor — every test in this module stays
 * network-free (no real server, no real socket, analogous to the C#
 * sibling's `TestClientFactory` + `FakeHttpMessageHandler`).
 */
internal object TestClientFactory {
    val serverAddress: URI = URI.create("http://example.invalid:8080")

    fun create(
        apiToken: String = "test-token",
        responder: (TransportRequest) -> TransportResponse,
    ): Pair<PgChangeFeedHttpClient, FakeHttpTransport> {
        val transport = FakeHttpTransport(responder)
        val options = PgChangeFeedClientOptions(serverAddress, apiToken)
        return PgChangeFeedHttpClient(transport, options) to transport
    }
}
